spec/app/dispatcher_spec.rb

```diff
@@ -127,6 +127,21 @@
           it_behaves_like 'logs exception'
         end
 
+        context 'when app exits unexpectedly' do
```

Comment:
Can you add a test where two SyncJobRunners are running, and both raise an Interrupt at some point, and ensure both get set with the correct failure status?